design/fast_finality.md

```diff
@@ -0,0 +1,299 @@
+# Fast finality design
+
+## Scope
+
+The introduction of a single sequencer to improve cost-per-transaction during the bootstrapping phase, as 
+described in the [Bootstrapping Strategy design doc](./Bootstrapping_strategy.md).
+
+## Glossary
+
+* *Light batch*: A set of transactions produced by the designated sequencer, considered to be soft-finalised in that 
+  order
+
+## Requirements
+
+* Finality
+  * Transaction *soft* finality (finality guaranteed by the sequencer) has a median duration of one second, from when 
+    the user submits the transaction to when the user receives the transaction receipts and events
+  * There is eventual transaction *hard* finality (finality guaranteed by the L1)
+  * The sequencer is strongly incentivised to hard-finalise transactions in the same order they are soft-finalised, and 
+    on the agreed cadence
+  * Nodes can easily prove whether a given soft-finalised transaction was hard-finalised, and in the correct order
+  * The sequencer is not able to "rewrite history" (or is strongly disincentivised from doing so), even for soft-final
+    transactions
+  * Soft finality does not break the mechanism used by some smart contracts (e.g. SushiSwap) of using `block.number` as 
+    a rudimentary clock
+* Censorship resistance
+  * End-users have a mechanism to force the sequencer to include their transactions (possibly at higher cost and slower 
+    finality)
+  * The sequencer distributes all soft-finalised transactions to all nodes
+* Value-extraction resistance
+  * The sequencer cannot precompute the effects of running a given set of transactions without committing to that set 
+    of transactions (e.g. by running a single transaction, then `eth_call`ing to see how it has affected a smart 
+    contract's state)
+* Cost
+  * L1 transaction costs can be driven lower at the expense of extending the hard-finality window
+  * The network can be configured to only publish rollups to the L1 every `x` blocks, where `x` >> 1
+  * The network can be configured to produce light batches (see below) at a higher frequency than L1 blocks
+* User/dev experience
+  * The responses to RPC calls reflect the soft-finalised transactions, and not just the hard-finalised transactions
+* Resilience
+  * Failover of the sequencer does not require a governance action
+  * During failover of the sequencer, it is acceptable to break the one-second soft-finality guarantee, and drop 
+    transactions that have not been soft-finalised
```

Comment:
another requirement to add:
- in case of an L1 reorg, the sequencer should make sure to republish the rollup exact rollup to mainain the order